Source/WebCore/dom/ElementRareData.h

```diff
@@ -98,6 +98,9 @@ class ElementRareData : public NodeRareData {
     RenderStyle* displayContentsStyle() const { return m_displayContentsStyle.get(); }
     void setDisplayContentsStyle(std::unique_ptr<RenderStyle> style) { m_displayContentsStyle = WTFMove(style); }
 
+    RenderStyle* displayNoneStyle() const { return m_displayNoneStyle.get(); }
+    void setDisplayNoneStyle(std::unique_ptr<RenderStyle> style) { m_displayNoneStyle = WTFMove(style); }
```

Comment:
`display:none` and `display:contents` are mutually exclusive. You don't need a separate field, you can just rename `m_displayContentsStyle` to `m_displayContentsOrNoneStyle` (along with the accessor functions).